x-pack/legacy/plugins/monitoring/index.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resolve } from 'path';
+import { Server } from 'src/legacy/server/kbn_server';
+import { i18n } from '@kbn/i18n';
+import { get } from 'lodash';
+
+import { LegacyPluginApi, LegacyPluginSpec, ArrayOrItem } from 'src/legacy/plugin_discovery/types';
+
+// @ts-ignore
+import { config as configDefaults } from './config';
+// @ts-ignore
+import { deprecations } from './deprecations';
+// @ts-ignore
+import { Plugin } from './server/plugin';
+// @ts-ignore
+import { initInfraSource } from './server/lib/logs/init_infra_source';
+
+type InfraPlugin = any; //TODO?
+type KibanaServer = { //TODO?
+  config: {
+    get: (key: string) => any
+  }
+  version: string
+  status: any
+}
+
+const uiExports = {
+  app: {
+    title: i18n.translate('xpack.monitoring.stackMonitoringTitle', { defaultMessage: 'Stack Monitoring' }),
+    order: 9002,
+    icon: 'plugins/monitoring/icons/monitoring.svg',
+    euiIconType: 'monitoringApp',
+    main: 'plugins/monitoring',
+  },
+  injectDefaultVars(server: Server) {
+    const config = server.config();
+    return {
+      isOnCloud: get(server.plugins, 'cloud.config.isCloudEnabled', false),
```

Comment:
we just added the client config for this https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#core-services check `uiExports/injectedVars` and example https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#configure-plugin